Fix 64

anhtr opened the pull request:
Hello,

This pull request is an attempt to fix #64. The first step is to update getStats to take more than one Comparison groups in the setting, and to create a dataframe with a "comp_grp" column for future grouping/filtering in the plotting. The second step is to update volcanoPlot to make use of the new dataframe to plot side by side plots.

Sorry for the large amount of "changes". Most of them were because I used "Reformat code" in RStudio.

Thank you.

jwildfire committed:
add linked table. #15

jwildfire committed:
remove custom callback

jwildfire committed:
remove count for now

Commit:
add soc/pt dropdown

Commit:
update plotly_click to show the correct soc in listing

anhtr committed:
Update getStats so that it takes more than one comparison group

anhtr committed:
Update getStats so that it takes more than one comparison group

anhtr committed:
Update volcanoPlot for synchonized subplots

anhtr committed:
Update volcanoPlot for synchonized subplots

anhtr committed:
Update so that VolcanoPlot only shows 1 plot if one comparison group

anhtr committed:
Update so that VolcanoPlot only shows 1 plot if one comparison group

anhtr commented:
(As of [5bd3186](https://github.com/SafetyGraphics/volcanoPlot/pull/69/commits/5bd3186f241e4457515f5c63ca1c2dac5bc67490))

So I implement this with crosstalk. I think it work decently well: two plots will be shown if two comparison groups are provided. If you click on a point in one graph, it will highlight the same point in the other. A thing I could not get it to do is to show the hover text on both graphs at the same time. However, you can make a compromise by clicking on one point in one graph and hover over the highlighted one in the other.

A current bug is that this is considered two plots, and currently would give an error if you run it in shiny app. I think either ui or server needs to be updated to handle the new output.

Test with this code:

```
settings<-list(
  stratification_col="AEBODSYS",
  group_col="ARM",
  reference_group="Placebo",
  comparison_group=c("Xanomeline High Dose","Xanomeline Low Dose"),
  id_col="USUBJID"
)
stats<-getStats(dfAE=safetyData::adam_adae, dfDemog = safetyData::adam_adsl, settings)
volcanoPlot(stats)
```

![image](https://user-images.githubusercontent.com/22552876/190505865-d1fe1410-7e0b-4f76-9e5f-435efcd3104a.png)

anhtr commented:
I also removed the option to not use plotly, because it doesn't make sense anymore.

anhtr commented:
Also for the next update: label the graphs with the comparison groups. I will work on it later and it should be simple enough that it won't break everything.

anhtr committed:
add commenting to volcanoPlot code

anhtr committed:
add commenting to volcanoPlot code

Commit:
work on filtering participant ids

Commit:
clean up

bzkrouse commented on R/volcanoPlot.R:
```diff
@@ -14,18 +13,19 @@
 #'
 #' @import ggplot2
 #' @importFrom plotly ggplotly
+#' @import crosstalk
```
```suggestion
#' @importFrom crosstalk SharedData bscols
```

bzkrouse replied:
Only importing what we need will help us to avoid the following warning:

```Warning message:
replacing previous import ‘crosstalk::getDefaultReactiveDomain’ by ‘shiny::getDefaultReactiveDomain’ when loading ‘volcanoPlot’ ```

bzkrouse commented on R/getStats.R:
```diff
@@ -27,70 +27,87 @@ getStats <- function(dfAE, dfDemog, settings, stat="Risk Ratio") {
     print(settings)
```
```suggestion
```

bzkrouse commented on R/volcano_server.R:
```diff
@@ -38,15 +38,54 @@ volcano_server <- function(input, output, session, params) {
+        # subset to selected strata
+        sub1<-all[all[params()$settings$aes$bodsys_col]==stats_sub()$strata,]
+
+        # subset to selected groups
```
Is it okay to delete this commented out code?

bzkrouse reviewed (changes requested):
@anhtr Great progress! I added some comments in the PR.
As you mentioned, there is a need to add the option to toggle the comparison group to the UI. We can discuss this further in our meeting, but here are some initial thoughts:

It looks like volcanoApp() receives information about the treatment groups from the settings object (currently the expectation is there will be a "group1" (reference group) and "group2" (comparison group) - see lines 24 & 25 of volcano_server.R.  

If we keep the settings object as-is, the user could set settings$group2 to be a vector of multiple choices. These would then populate a dropdown in the UI. The user selection would then be passed to line 25 in the volcano_server. 
However I might suggest we rename group1 and group2 to be something more informative.

-----------------------------------------------------------------------------------------------------
To recap group discussion:
- No UI dropdown needed
- settings to specify the reference group but just assume the rest of the trt groups will be comparison groups
- Display all comparisons

nandriychuk committed:
Merge pull request #68 from SafetyGraphics/add-soc-pt

add soc/pt dropdown

jwildfire committed:
Merge pull request #67 from SafetyGraphics/fix-58

add linked table. #15

anhtr committed:
Merge remote-tracking branch 'origin/fix-64' into fix-64

anhtr committed:
Showing dynamic number of plots

anhtr committed:
Remove unused EnhancedVolcano plot

anhtr committed:
Group2 contains everything but 'Placebo'

anhtr committed:
volcanoPlot() now pull group label directly out of data

anhtr committed:
Update so that clicking point shows table with multiple plots

anhtr committed:
add commenting to clarify the codes

anhtr commented:
I have updated the package:

- If setting not specified, the default reference group is 'Placebo' and everything else will be comparison group. As the result:
  + getStats() was updated to put group names directly in the output table.
  + volcanoPlot() pulls group names directly out of the stats table.
  + Anything that uses groups() reactive is now updated to stop using it, as it does not make sense anymore.
  + We no longer filter for groups in the click table.
- Arbitrary number of plots should display correctly in the UI.
- EnhancedVolcano plot was removed. https://github.com/SafetyGraphics/volcanoPlot/issues/70
- Minimal comments were added to clarify functions.

anhtr committed:
minor changes to what get printed out in console

anhtr committed:
Correct getstats() stat argument

jwildfire committed:
Update R/volcanoPlot.R

Co-authored-by: Becca Krouse <becca.krouse@gmail.com>

jwildfire commented:
Closing this one since changes are captured in #74 also
